pom.xml

```diff
@@ -31,6 +31,35 @@
             <artifactId>spring-boot-starter-test</artifactId>
             <scope>test</scope>
         </dependency>
+
+        <dependency>
+            <groupId>com.h2database</groupId>
+            <artifactId>h2</artifactId>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-data-jpa</artifactId>
+            <version>3.1.2</version>
```

Comment:
remove all spring-boot-starter versions - you have 
        <artifactId>spring-boot-starter-**parent**</artifactId>
        <version>3.1.1</version>